aiohttp/abc.py

```diff
@@ -145,3 +145,18 @@ def __init__(self, logger, log_format):
     @abstractmethod
     def log(self, request, response, time):
         """Emit log to logger."""
+
+
+class AbstractRuleMatching(ABC):
```

Comment:
I doubt if we need this abstract class: it is used for `Domain` class only.

Reply:
Still the question

Reply:
There are two and a half ways to decide his fate:

1 Throw away

2 To refine the subsystem of rules

2+1/2. Combine MatchedSubAppResource and PrefixedSubAppResource as they differ practically only by the method resolve

    class PrefixedSubAppResource(PrefixResource):
        async def resolve(self, request):
            if not request.url.raw_path.startswith(self._prefix):
                return None, set()

    class MatchedSubAppResource(PrefixedSubAppResource):
        async def resolve(self, request):
            if not await self._rule.match(request):
                return None, set()


and implement the rule

    class Prefix(AbstractRuleMatching):
        def __init__(self, prefix):
            self._prefix = prefix
    
        @property
        def prefix(self):
            return self._prefix
        
        def match(self, request):
            return request.url.raw_path.startswith(self._prefix)

Reply:
I'd like to not introduce a new **public** ABC class.
Moreover, in aiohttp 4.0 I want to drop a pluggable router. By this, we'll have the only supported router implementation -- the current one. In this light adding a new ABC to API part that will never be extended outside doesn't make any sense.

If you have a feeling that ABC is valuable -- feel free to keep it but move into `web_urldispatcher.py`.